offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -23,112 +26,165 @@ export class CkbHandler {
-          );
-        }
-      }
-      await this.onBurnTxs(latestHeight, burnTxs);
+      await this.onBlock(block);
+    }
+  }
 
-      latestHeight++;
-      await asyncSleep(1000);
+  async onBlock(block: Block) {
+    const blockNumber = Number(block.header.number);
+    const blockHash = block.header.hash;
+    logger.info(`CkbHandler onBlock blockHeight:${blockNumber} blockHash:${blockHash}`);
+
+    const confirmNumber = ForceBridgeCore.config.ckb.confirmNumber;
+    const confirmedBlockHeight = blockNumber - confirmNumber >= 0 ? blockNumber - confirmNumber : 0;
+    if (
+      confirmNumber !== 0 &&
+      this.lastHandledBlockHeight === blockNumber - 1 &&
+      this.lastHandledBlockHash !== '' &&
+      block.header.parentHash !== this.lastHandledBlockHash
+    ) {
+      logger.warn(
+        `CkbHandler onBlock blockHeight:${blockNumber} parentHash:${block.header.parentHash} != lastHandledBlockHash:${this.lastHandledBlockHash} fork occur removeUnconfirmedLock events from:${confirmedBlockHeight}`,
+      );
+      await this.db.removeUnconfirmedCkbBurn(blockNumber, confirmNumber);
+
+      const confirmedBlock = await this.ckb.rpc.getBlockByNumber(BigInt(confirmedBlockHeight));
+      await this.setLastHandledBlock(Number(confirmedBlock.header.number), confirmedBlock.header.hash);
+      return;
+    }
+
+    await this.setLastHandledBlock(blockNumber, blockHash);
+    const unconfirmedTxs = await this.db.getUnconfirmedCkbBurnToConfirm(blockNumber, confirmNumber);
+    const confirmedTxHashes = unconfirmedTxs.map((burn) => {
+      return burn.ckbTxHash;
+    });
+    await this.db.updateCkbBurnConfirmStatus(confirmedTxHashes);
+    await this.onCkbBurnConfirmed(unconfirmedTxs);
+
+    const burnTxs = new Map();
+    for (const tx of block.transactions) {
+      if (await this.isMintTx(tx)) {
+        await this.onMintTx(tx);
+      }
+      const recipientData = tx.outputsData[0];
+      let cellData;
+      try {
+        cellData = new RecipientCellData(fromHexString(recipientData).buffer);
+      } catch (e) {
+        continue;
+      }
+      if (await this.isBurnTx(tx, cellData)) {
+        const burnPreviousTx: TransactionWithStatus = await this.ckb.rpc.getTransaction(
+          tx.inputs[0].previousOutput.txHash,
+        );
+        const senderLockHash = this.ckb.utils.scriptToHash(
+          burnPreviousTx.transaction.outputs[Number(tx.inputs[0].previousOutput.index)].lock,
+        );
+        const data: BurnDbData = {
+          senderLockScriptHash: senderLockHash,
+          cellData: cellData,
+        };
+        burnTxs.set(tx.hash, data);
+        logger.info(
+          `CkbHandler watchBurnEvents receive burnedTx, ckbTxHash:${
+            tx.hash
+          } senderLockHash:${senderLockHash} cellData:${JSON.stringify(cellData, null, 2)}`,
+        );
+      }
     }
+    await this.onBurnTxs(blockNumber, burnTxs);
```

Comment:
L194  ~ L198

```
    const pendingMintTxs = await this.db.getMintRecordsToUpdate(tx.hash);
    pendingMintTxs.map((r) => {
      r.status = 'success';
    });
    await this.db.updateCkbMint(pendingMintTxs);
```

better be replaced with `update` sql to reduce db request.

Reply:
fixed